jwst/associations/asn_edit.py

```diff
@@ -165,4 +166,67 @@ def _path(filename):
         Full path including any username and environment
         variables included.
     """
-    return Path(op.expandvars(filename)).expanduser().resolve()
+    return Path(os.path.expandvars(filename)).expanduser().resolve()
+
+
+def writer(asn, output_file):
+    """
+    Write the association out to disk.
+
+    Parameters
+    ----------
+    asn : object
+        An association object
+
+    output_file : str or Path
+        The filename of the association
+
+    Raises
+    ------
+    ValueError
+        The output filename was not a json file
+    """
+    output_file = Path(output_file)
+    asn_format = output_file.suffix[1:]
+    if asn_format != "json":
+        raise ValueError("Only json format supported for output: " + output_file)
+
+    serialized = json.dumps(asn, indent=4, separators=(",", ": "))
+
+    in_place = Path.exists(output_file)
+    if in_place:
+        temp_file = _rename(output_file)
+
+    try:
+        fd = Path.open(output_file, "w")
+        fd.write(serialized)
+        fd.close()
```

Comment:
Context manager might be safer so `fd` won't be left dangling when bad things happen. ref: https://docs.python.org/3/library/pathlib.html#reading-and-writing-files

```suggestion
        with Path(output_file).open("w") as fd:
            fd.write(serialized)
```

Otherwise, please consider moving `fd.close()` to a `finally:` block. Thanks!